tests/test_cacheable_pdf.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+import zfit
+import zfit.z.numpy as znp
+import tensorflow as tf
+from zfit.models.cache import CacheablePDF
+
+mu = zfit.Parameter("mu", 1.0, -5, 5)
+sigma = zfit.Parameter("sigma", 1, 0, 10)
+obs = zfit.Space("x", limits=[-5.0, 5.0])
+gauss = zfit.pdf.Gauss(mu=mu, sigma=sigma, obs=obs)
+cached_gauss = CacheablePDF(gauss)
+x = znp.linspace(-5, 5, 500)
+
+
```

Comment:
Also, extend the tests. We want to check if it actually caches and if it still works when the parameters are changed